src/notification/NotificationController.ts

```diff
@@ -0,0 +1,111 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+
+interface viewedNotification {
+  [id: number]: boolean;
+}
+
+interface Notification {
+  id: number;
+  title: string;
+  description: string;
+  date: string;
+  image?: string;
+  actionText?: string;
+}
+
+interface StateNotification extends Notification {
+  isShown: boolean;
+}
+
+/**
+  * A map of notification ids to Notification objects
+  */
+interface NotificationMap {
+  [id: number]: Notification;
+}
+
+/**
+  * A map of notification ids to StateNotification objects
+  */
+export interface StateNotificationMap {
+  [id: number]: StateNotification;
+}
+
+/**
+ * NotitificationConfig will hold the notifications from JSON file read
+ * from `metamask-extension`
+ */
+export interface NotificationConfig extends BaseConfig{
+  allNotifications: NotificationMap;
+}
+
+/**
+ * Notification state will hold all the seen and unseen notifications
+ * that are still active
+ */
+export interface NotificationState extends BaseState{
+  notifications: StateNotificationMap;
+}
+
+const defaultState = {
+  notifications: {},
+};
+
+/**
+ * Controller for managing in-app announcement notifications.
+ */
+export class NotificationController extends BaseController<NotificationConfig, NotificationState> {
+
+  private readonly allNotifications: NotificationMap;
+
+  /**
+   * Creates a NotificationController instance
+   *
+   * @param config - Initial options used to configure this controller
+   * @param state - Initial state to set on this controller
+   */
+  constructor(config: NotificationConfig, state?: NotificationState) {
+    const { allNotifications } = config;
+    super(config, state || defaultState);
+    this.allNotifications = { ...allNotifications };
```

Comment:
It seems like the notifications are being stored _twice_ - once as part of the config, and a second time here on this private property. Could we use the config instead, to avoid duplicating it?

Reply:
It means using config instead of storing it in the private property, right?!

Reply:
Exactly, yep!